### PR TITLE
FEATURE: CQ-582 Semantic Search SDK component

### PR DESCRIPTION
**Pull request by andriichumak**
A new package is created for semantic search: `@gooddata/sdk-ui-semantic-search`. The reason for a new package is because it's a "smart" component and it has to be embeddable by the end user eventually.

Unit tests and simple e2e added, need to record screenshots with backstop.

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

**Comment by andriichumak** [comment]
extended test - backstop

File: libs/sdk-ui-semantic-search/config/rush-project.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "$schema": "https://developer.microsoft.com/json-schemas/rush/v5/rush-project.schema.json",
+    "incrementalBuildIgnoredGlobs": [],
+    "operationSettings": [
+        {
+            "operationName": "build",
+            "outputFolderNames": ["esm"]
```
Comment (ivanmjartan) [inline libs/sdk-ui-semantic-search/config/rush-project.json:7]: I guess that styles/scss is missing in rush cache definition see: https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-ui-kit/config/rush-project.json#L7

File: libs/sdk-ui-semantic-search/package.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        "directory": "libs/sdk-ui-semantic-search"
+    },
+    "author": "GoodData Corporation",
+    "license": "MIT",
+    "type": "module",
+    "browser": "./esm/index.js",
+    "exports": {
+        ".": "./esm/index.js",
+        "./styles/css/*": "./styles/css/*",
+        "./internal": "./esm/internal.js"
```
Comment (ivanmjartan) [inline libs/sdk-ui-semantic-search/package.json:17]: Do you need internal export? Pls remove it and export only main entry point.

File: libs/sdk-ui-semantic-search/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// (C) 2020-2024 GoodData Corporation
+module.exports = {
+    parser: "@typescript-eslint/parser",
+    plugins: ["react-hooks", "prettier", "sonarjs", "eslint-plugin-tsdoc"],
+    extends: [
+        "@gooddata",
```
Comment (ivanmjartan) [inline libs/sdk-ui-semantic-search/.eslintrc.js:6]: Pls can you verify (nejsem si jisty) that eslint is set check hooks and has set it as error (warnings are deafult)

        "react-hooks/rules-of-hooks": 2,
        "react-hooks/exhaustive-deps": 2,

**Comment by andriichumak** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2683/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by andriichumak** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2685/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Review by ivanmjartan — APPROVED** [review]

**Pull request merged**